compiler/codegen/cg-expr.cpp

```diff
@@ -1023,6 +1023,23 @@ static GenRet codegenRlocale(GenRet wide)
   return ret;
 }
 
+static GenRet codegenRsubloc(GenRet wide){
+  GenRet ret;
+  Type* type = SUBLOC_ID_TYPE;
+
+  if( !fLLVMWideOpt ) {
+    ret = codegenCallExpr("chpl_sublocFromLocaleID",
+                          codegenValue(
+                              codegenWideThingField(wide, WIDE_GEP_LOC)),
+                          /*ln*/codegenZero(), /*fn*/codegenZero32());
+  } else {
+    USR_FATAL("--llvm-wide-opt can't be used with the GPU locale model");
```

Comment:
See https://bpb-us-e1.wpmucdn.com/blogs.rice.edu/dist/1/2385/files/2013/07/Chapel_LLVM_camera_ready-q6usv4.pdf. I think it is a nice optimization that I wish to make a default one. It may also help GPUs nicely. Though it has some issues: https://github.com/Cray/chapel-private/issues/2121. We'd need to fix them first before making it work here.